src/python/WMCore/WorkerThreads/BaseWorkerThread.py

```diff
@@ -225,6 +233,10 @@ def __call__(self, parameters):
         msg = "Worker thread %s terminated" % str(self)
         logging.info(msg)
 
+        # send heartbeat message that thread is terminated
+        if self.useHeartbeat and self.heartbeatAPI is not None:
```

Comment:
Ah-haa, I like it! However, it will mask the real exception in the algorithm, see:
https://github.com/ticoann/WMCore/blob/88b40afcdad9bcaef291888fa1b0f88b623468e7/src/python/WMCore/WorkerThreads/BaseWorkerThread.py#L212

Reply:
Good point, the problem is there are other exceptions than that code path could happen and we don't seem to report that. I will try to improved to handle this.